### PR TITLE
VMware: Add new module: vmware_host_config_manager

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py
test/integration/targets/vmware_host_config_manager/aliases
test/integration/targets/vmware_host_config_manager/tasks/main.yml

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
Integration test missing due to https://github.com/vmware/govmomi/issues/989

File: lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    - If C(cluster_name) is not given, this parameter is required.
+  options:
+    description:
+    - A dictionary of advance configuration parameter.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather facts about all ESXi Host in given Cluster
+  vmware_host_config_facts:
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: `vmware_host_config_facts` → `vmware_host_config_manager`.

File: lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+- Abhijeet Kasurde (@akasurde)
+notes:
+- Tested on vSphere 6.5
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  cluster_name:
+    description:
+    - Name of the cluster from which the ESXi host belong to.
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: How the ESXi is selected if we only set `cluster_name` and there is multiple ESXi in the cluster ?

Comment (tchernomax) [reply lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: I just read the test.

> Change facts about all hosts in given cluster

You should change the `description` of `cluster_name`.'

File: lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    - If C(esxi_hostname) is not given, this parameter is required.
+  esxi_hostname:
+    description:
+    - ESXi hostname to gather facts from.
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: As I understand it, this module isn't about gathering facts.

**Review by tchernomax — CHANGES_REQUESTED** [review]
Except from the documentation part, this PR look good to me.

File: lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+  options:
+    description:
+    - A dictionary of advance configuration parameter.
+    - Invalid configuration parameters are ignored.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Manage Log level setting for all ESXi Host in given Cluster
+  vmware_host_config_facts:
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: `vmware_host_config_facts` → `vmware_host_config_manager`

File: lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: cluster_name
+    options:
+        'Config.HostAgent.log.level': 'info'
+
+- name: Manage Log level setting for an ESXi Host
+  vmware_host_config_facts:
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: `vmware_host_config_facts` → `vmware_host_config_manager`

File: lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    options:
+        'Config.HostAgent.log.level': 'verbose'
+
+- name: Manage multiple settings for an ESXi Host
+  vmware_host_config_facts:
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_config_manager.py]: `vmware_host_config_facts` → `vmware_host_config_manager`

**Review by tchernomax — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
@tchernomax Could you please take a look again ?

**Review by tchernomax — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@tchernomax Thanks for your review.

**Pull request merged**